R/data.R

```diff
@@ -105,6 +105,72 @@ process_data <- function(data) {
   path
 }
 
+#' Process fitted params for the generate quantities method
+#'
+#' @noRd
+#' @param data A path to a CSV file compatible with CmdStan or a sampling fit object (CmdstanMCMC).
+#' @return Path to data file.
+process_fitted_params <- function(fitted_params) {
+  if (is.character(fitted_params)) {
+    paths <- absolute_path(fitted_params)
+  } else if (checkmate::test_r6(fitted_params, classes = ("CmdStanMCMC"))) {
+    if (all(file.exists(fitted_params$output_files()))) {
+      paths <- absolute_path(fitted_params$output_files())
+    } else {
+      draws <- tryCatch( posterior::as_draws_array(fitted_params$draws()),
+        error=function(cond) {
+            stop("Unable to obtain draws from the fit (CmdStanMCMC) object.", call. = FALSE)
+        }
+      )
+      sampler_diagnostics <- tryCatch(posterior::as_draws_array(fitted_params$sampler_diagnostics()),
+        error=function(cond) {
+            stop("Unable to obtain sampler diagnostics from the fit (CmdStanMCMC) object.", call. = FALSE)
+        }
+      ) 
+      if (!is.null(draws)) {
+        variables <- dimnames(draws)$variable
```

Comment:
Ah yes. I keep forgetting on that one! Thanks.

Reply:
This particular case can be replaced because it will only urn for non-null draws. For others, we need to fix https://github.com/stan-dev/posterior/issues/78